### DISCUSSION TITLE
[python-pytorch-rocm] question about gfx1031 (67000 xt)

### DISCUSSION THREAD

**Discussion by fractal-fumbler** [rocm-arch/rocm-arch#895] [Support] [answered]
hello, folks :)

with help of DorianRudolph's PR (https://github.com/rocm-arch/rocm-arch/pull/836) i've made pytorch to work. Took a long tho to compile :D

but after recent release of rocm 5.3.0 i am getting error `"hipErrorNoBinaryForGpu: Unable to find code object for all current devices!"` 

i've recompiled all needed packages with AMDGPU_TARGETS="gfx1030;gfx1031" and pytorch with PYTORCH_ROCM_ARCH="gfx1030;gfx1031"
(expection is rccl with error ```lld: error: ld-temp.o <inline asm>:1:2: instruction not supported on this GPU                                           
        buffer_wbinvl1_vol                                                                                              
        ^                   ```
how to find out what package is responsible for missing binaries? :)

**Comment by fractal-fumbler** [comment, marked as answer]
so i've downgraded rccl to 5.2.3 and kind of success :D 
```python
 python -c 'import torch; print(torch.cuda.is_available())'
True
```

**Reply by tpkessler** [reply]
The reason for this is that I removed the option to set the AMDGPU_TARGET as part of my effort to bring the ROCm packages to the official Arch Linux repos. In version 5.3 rccl officially supports only gfx1030 but not gfx1031, see https://github.com/ROCmSoftwarePlatform/rccl/blob/a44e92c879e2d2aff2ffcbc5b57a9862728ca92d/CMakeLists.txt#L37

**Reply by fractal-fumbler** [reply]
yes, i do remember about your efforts to bring to community and appreciate it :)
thanks for your work!

as i understand one need some kind of patch to get gfx1031 to rccl?

**Reply by tpkessler** [reply]
Download the PKGBUILD of rccl and add `-DAMDGPU_TARGETS="gfx1030;gfx1031"` to the cmake arguments.
